File: Data Structures & Algorithms/valid-binary-search-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:

        l = []
        def in_order(root: Optional[TreeNode]):
            
            if root is None:
                return

            in_order(root.left)
            l.append(root.val)
            in_order(root.right)
        
        in_order(root)

        first = l[0]
        for i in range(1, len(l)):  

            if l[i] > first:
                first = l[i]
            else:
                return False
        
        return True

           





            


        
        

File: Data Structures & Algorithms/is-palindrome/submission-3.cpp
class Solution {
public:
    bool isPalindrome(string s) {

        int left = 0;
        int right = s.size() -1;

        for (int left = 0, right = s.size()-1; left<right;){
            if (std::isalnum(s[left]) && std::isalnum(s[right])){

                if(char(tolower(s[left])) == char(tolower(s[right])) ){
                    right--;
                    left++;
                } else {
                    return false;
                }

            } else if (!(std::isalnum(s[left]))){
                left++;
            }else{
                right--;
            }
        }

        return true;
        
    }
};
